app/src/main/java/com/coronavirus/data/WebServiceProvider.kt

```diff
@@ -0,0 +1,25 @@
+package com.coronavirus.data
+
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.okhttp.OkHttp
+import io.ktor.client.features.json.GsonSerializer
+import io.ktor.client.features.json.JsonFeature
+import io.ktor.client.request.get
+
+class WebServiceProvider {
+
+  val client: HttpClient =
+    HttpClient(OkHttp) {
+      install(JsonFeature) {
+        serializer = GsonSerializer()
+      }
+    }
+
+  suspend inline fun <reified T> get(urlContext: String): T {
```

Comment:
do we need the reified here? i don't the generic type access, but maybe  ktor needs it 🤷

Reply:
It's necessary because inside de get function we need the generic type when we do `client.get<T>`.  It's true, ktor HttpClient get extension function need it!